### ISSUE TITLE
What is the relation between inertia force and others in wec-sim?

### ISSUE THREAD

**Issue by zzq1019** [WEC-Sim/WEC-Sim#573]
In the simulation of RM3 project, I want to calculate the inertia force: ma. the simu.b2b = 0.  From my understanding:
M = diag([body(1,1).mass, body(1,1).mass, body(1,1).mass, body(1,1).momOfInertia])
a = output.bodies(1).acceleration';
Ma = output.bodies(1).forceTotal + output.ptos.forceToal
At t=0, the ptoForce = 0, so the ma of body1 is equal to the forceTotal of body1.
At least in z-axis, I think the above formulations are correct. However, the fact is not.
So, what is the real relation between inertia force and others?
I look forward to your response. Thanks in advance!
![image](https://user-images.githubusercontent.com/45420295/116769181-d617b580-aa6c-11eb-88fb-dfb611d1a740.png)
![image](https://user-images.githubusercontent.com/45420295/116769184-da43d300-aa6c-11eb-8f9c-8e33f15de50f.png)

![image](https://user-images.githubusercontent.com/45420295/116769188-e039b400-aa6c-11eb-8e6e-8bf177f39515.png)
![image](https://user-images.githubusercontent.com/45420295/116769191-e596fe80-aa6c-11eb-9921-3b54ebbf8528.png)

![image](https://user-images.githubusercontent.com/45420295/116769192-eb8cdf80-aa6c-11eb-8b48-2e8aeed65389.png)
![image](https://user-images.githubusercontent.com/45420295/116769196-efb8fd00-aa6c-11eb-9fa5-f2161d5cf932.png)

**Comment by kmruehl** [comment]
@zzq1019  Can you please clarify this request? Are you trying to calculate the inertial force as F=ma?

**Comment by zzq1019** [comment]
Thanks @kmruehl for the reply.
Sorry for my poor english. And, yes, I am trying tio calculate the inertial force using the simulation results:`output`
The equation of motion for a floating body can be given as:
![image](https://user-images.githubusercontent.com/45420295/117232594-0b991600-ae54-11eb-8284-a895fad9accc.png)

In the simulation of RM3, I want to verify the simmulation results about the equation above. The `wecSimInputFile.m` as:
![image](https://user-images.githubusercontent.com/45420295/117232537-f02e0b00-ae53-11eb-9889-cc0121025bba.png)
![image](https://user-images.githubusercontent.com/45420295/117232566-fcb26380-ae53-11eb-8267-3c2a583baa59.png)


From my understanding, for the body(1):
`F_ext + F_rad + F_B = output.bodies(1).forceExcitation - output.bodies(1).forceRadiationDamping -output.bodies(1).forceAddedMass - forceRestoring = output.bodies(1).foreTotal`
`F_v = 0, F_me = 0, F_m = 0`
`F_pto = output.bodies(1).forceTotal`

From other issuses, I have learned that all the forces act on the center of the object. So, in addition to the Z-axis motion, the above equation is not valid (I don't know if my understanding is correct). 
At least in Z-axis of body(1), it bears only two forces: the `output.bodies(1).forceTotal` of z-axis and `output.ptos.forceTotal` of z-axis. So, the equation of motion for body(1) shoul as: 
`body(1,1).mass * output.bodies(1).acceleration(:,3) = output.bodies(1).forceTotal(:,3) + output.ptos.forceTotal(:,3)`

In particular at time `t = 0`,
`output.ptos.forceTotal(:,3) = 0`
 `body(1,1).mass * output.bodies(1).acceleration(:,3) = output.bodies(1).forceTotal(:,3)`

In the simulation results, the above equation does not hold.
Is there anything obvious I might be missing here? 
Thanks

**Comment by yuyihsiang** [comment]
Were you running the RM3 with 3DOFs? If so, I think the formulation you have won't held because output.ptos.forceTotal(:,3) is actually the translational force between the two bodies, not the heave force. You may want to test the same formulation by constraining the RM3 motion to heave only (i.e., replace the 3DOF block with a translational block) to test it out. We are definitly interested in the outcome.

Thanks,

--Yi-Hsiang

**Comment by zzq1019** [comment]
Hi @yuyihsiang 
I know what you mean, the direction of `output.ptos.forceTotal(:,3)` is  along the center of mass of body(1) and body(2). And I have run the RM3 with translational block.
![image](https://user-images.githubusercontent.com/45420295/118352261-0db94e00-b593-11eb-8766-bca1e16fe065.png)

And we can see the position of body(1), it is only moving in the z direction.
![image](https://user-images.githubusercontent.com/45420295/118352820-11020900-b596-11eb-81e4-739db5ea0780.png)

The results are shown in the figures. 
 
![image](https://user-images.githubusercontent.com/45420295/118353515-2cbade80-b599-11eb-8664-6433ced2f95d.png)
![image](https://user-images.githubusercontent.com/45420295/118353523-40664500-b599-11eb-8cbb-8c8184b256cd.png)
![image](https://user-images.githubusercontent.com/45420295/118353571-802d2c80-b599-11eb-9742-49cb5a3394d1.png)

`body(1,1).mass = 725834`
`body(1,1).mass * output.bodies(1).acceleration(:,3)`:
![image](https://user-images.githubusercontent.com/45420295/118353623-be2a5080-b599-11eb-9fd9-d74b62a7a0dd.png)

`output.bodies(1).forceTotal(:,3) + output.ptos.forceTotal(:,3)`
![image](https://user-images.githubusercontent.com/45420295/118353707-1f522400-b59a-11eb-897c-ece259c6a908.png)

So, we can see that `body(1,1).mass * output.bodies(1).acceleration(:,3)` is not equal `output.bodies(1).forceTotal(:,3) + output.ptos.forceTotal(:,3)`

Thanks

**Comment by yuyihsiang** [comment]
Thank you for testing it. I also looked into this and got something similar to what you have shown here. 

If you plot the two forces in time for several seconds, you will see a small difference between the two. The difference we see here was caused by the use of simu.adjMassWeightFun and the lag of added-mass term in time on the right-hand side of the equation. Therefore, the output.bodies(1).forceTotal(:,3) we see here is not the exact total force solved in the equation of motion. I did a test by zeroing out the added-mass term in surge and sway to zero and adjusting the simu.adjMassWeightFun to 1, which forces the infinite frequency added mass scalar value equal to the heave added mass value, and the two forces become identical.

--Yi-Hsiang
